### PR TITLE
Fixes #179: Migration of current travis ci to github-actions ci.

### LINKED ISSUE #179: Swich to github actions for CI builds
Currently, we are using Travis for CI builds. We should remove third party app dependancy with Github actions.

### PR DESCRIPTION
**Pull request by ameydev**
##### ISSUE TYPE

 - Feature Pull Request

Fix for issue [githib-actions](https://github.com/infracloudio/botkube/issues/179)

##### SUMMARY

-  Updated the golint path to $GOPATH/bin/golint to fix github-actions build failure.
-  Added docker push action step in current CI steps.
-  Added helm release step along with missing "helm init".

Checklist for Enabling github actions:
make sure to add below secrets in repo/settings/secrets
- GH_TOKEN(github access token of **charts** repo)
- DOCKER_USERNAME and DOCKER_PASSWORD (for **infracloud/botkube** repo).
- That's it, by merging this PR, first github-actions workflow for botkube should be triggered :) 

For reference the tested CI workflows on my repos:

- [with helm release](https://github.com/ameydev/botkube/runs/434885674?check_suite_focus=true)
- [Normal workflow with docker push and helm release skipped](https://github.com/ameydev/botkube/runs/434903893?check_suite_focus=true)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+on:
+  push:
+  pull_request:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      GO111MODULE: on
+      GOPATH: /home/runner/work/botkube
+      GOBIN: /home/runner/work/botkube/bin
```
Comment (PrasadG193) [inline .github/workflows/ci.yml]: What is $HOME value? Can we use $HOME to set the path?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        cd ..
+        git config --global user.email "botkube-ci@infracloud.io"
+        git config --global user.name "BotKube CI"
+        git clone https://github.com/infracloudio/charts.git
+        cd charts
+        git fetch
+        git branch
+        git pull origin gh-pages
+        helm init --client-only
+        helm package --app-version v9.99.9-ci-test --version v9.99.9-dev ${{ github.workspace	}}/helm/botkube/
```
Comment (PrasadG193) [inline .github/workflows/ci.yml]: Please keep `--app-version` same as `--version` i.e v9.99.9-dev

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ameydev) [reply .github/workflows/ci.yml]: $HOME value is /home/runner. We can use that for $GOPATH.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ameydev) [reply .github/workflows/ci.yml]: yes sure, will update the change. I missed that

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ameydev) [reply .github/workflows/ci.yml]: the value for $HOME is /home/runner but it gives error if we try to assign value of GOPATH to a relative path. 
Check [error](https://github.com/ameydev/botkube/runs/435702976?check_suite_focus=true#step:4:11).
so I think we need to give absolute path which would  be `/home/runner` or can keep the same value.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ameydev) [reply .github/workflows/ci.yml]: Pushed the update!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        cd ..
+        git config --global user.email "botkube-ci@infracloud.io"
+        git config --global user.name "BotKube CI"
+        git clone https://github.com/infracloudio/charts.git
+        cd charts
+        git fetch
+        git branch
+        git pull origin gh-pages
+        helm init --client-only
+        helm package --app-version v9.99.9-dev --version v9.99.9-dev ${{ github.workspace	}}/helm/botkube/
```
Comment (PrasadG193) [inline .github/workflows/ci.yml]:
```suggestion
        helm package --app-version v9.99.9-dev --version v9.99.9-dev ${{ github.workspace }}/helm/botkube/
```

**Comment by PrasadG193** [comment]
Thanks for the PR @ameydev. LGTM

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ameydev) [reply .github/workflows/ci.yml]: Removed the extra spaces.

**Comment by ameydev** [comment]
> Thanks for the PR @ameydev. LGTM

Your welcome! just a reminder
Before merging:
Checklist for Enabling github actions:
make sure to add below secrets in repo/settings/secrets

GH_TOKEN(github access token of charts repo)
DOCKER_USERNAME and DOCKER_PASSWORD (for infracloud/botkube repo).

**Comment by PrasadG193** [comment]
@ameydev We should get this merged early in this release cycle. Could you please remove `.travis.yaml` as a part of this PR?

**Comment by ameydev** [comment]
@PrasadG193 I think we can keep travis and gh-actions for a while and once we make sure everything is working fine, can remove .travis.yaml.

**Pull request merged**